lib/pbench/test/functional/server/test_datasets.py

```diff
@@ -686,15 +723,25 @@ class TestUpdate:
     @pytest.mark.parametrize("access", ("public", "private"))
     def test_publish(self, server_client: PbenchServerClient, login_user, access):
         expected = "public" if access == "private" else "private"
-        datasets = server_client.get_list(access=access, mine="true")
+        datasets = server_client.get_list(
+            access=access, mine="true", metadata=["server.archiveonly"]
+        )
         print(f" ... updating {access} datasets to {expected} ...")
         for dataset in datasets:
-            server_client.update(dataset.resource_id, access=expected)
-            print(f"\t ... updating {dataset.name} to {access!r}")
-            meta = server_client.get_metadata(
-                dataset.resource_id, metadata="dataset.access"
+            response = server_client.update(
+                dataset.resource_id, access=expected, raise_error=False
             )
-            assert meta["dataset.access"] == expected
+            print(f"\t ... updating {dataset.name} to {access!r}")
+            if response.ok:
```

Comment:
Shouldn't the `print()` at line 734 precede the operation at line 731?  (And, shouldn't the `print()` be referencing `expected` rather than `access`, since that's what line 732 does?)

Reply:
This shows up as a diff because I rearranged code, but I didn't change the line.

Reply:
True.  But, I still think the lines are in the wrong order, etc.  🙂